python/ray/data/tests/test_object_gc.py

```diff
@@ -235,6 +234,29 @@ def consume(p):
     assert "Spilled" not in meminfo, meminfo
 
 
+def test_global_bytes_spilled(shutdown_only):
+    # The object store is about 90MB.
+    ctx = ray.init(object_store_memory=90e6)
+    # The size of dataset is 500*(80*80*4)*8B, about 100MB.
+    ds = ray.data.range_tensor(500, shape=(80, 80, 4), parallelism=100).materialize()
```

Comment:
Won't spilling occur when you call `materialize` here?

Reply:
Yes, this test checks for spills

Reply:
Ah, I guess I was wondering why we call `iter_batches` in `check_no_spill` then? Since the spilling has already occurred?

Out of scope for this PR, though. I understand that the other tests call `repeat` and this test is just following that pattern.